### PR TITLE
Fix typos in README.md

### PR DESCRIPTION
**Pull request by norihiro**
### Description

Fix some typos in the README.md file that are flagged by some spell-check tools.

Also removed trailing white spaces.

### Motivation and Context

Less number of typos is good.

### How Has This Been Tested?

Tested using these spell-check tools (not all errors highlighted by the tools were fixed but only obvious errors were fixed.).
- Gmail
- Grammarly
- LibreOffice 7.6.6.3

### Types of changes

- Documentation (a change to documentation pages)

### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,15 +141,15 @@ It is recommended that no work start on implementation until an RFC is accepted.
...
 will ultimately be merged; it does mean that in principle all the major interested
 parties have agreed to the feature and are amenable to merging it.
 
 Furthermore, the fact that a given RFC has been accepted and is "active"
 implies nothing about what priority is assigned to its implementation, nor does
-it imply anything about whether an OBS Project developer has been assigned the 
-task of implementing the feature. While it is not *necessary* that the author 
+it imply anything about whether an OBS Project developer has been assigned the
+task of implementing the feature. While it is not *necessary* that the author
 of the RFC also write the implementation, it is by far the most effective way
```
Comment (norihiro) [inline README.md:153]: I assumed `should` was omitted in the line 153, I didn't fix `write` to `writes`.

Comment (Fenrirthviti) [reply README.md:153]: I can see why spellcheck would want to make that change, but this reads correctly to me as a native english speaker. Adding "should" makes it read more awkward, IMO.

Comment (RytoEX) [reply README.md:153]: > While it is not *necessary* that the author of the RFC also write the implementation

This is grammatically correct, as far as I can tell.

**Review by Fenrirthviti — APPROVED** [review]
All changes look good to me.

**Review by RytoEX — APPROVED** [review]

**Pull request merged**